project/src/main/java/com/google/sps/servlets/SellInvestmentServlet.java

```diff
@@ -0,0 +1,124 @@
+package com.google.sps.servlets;
+
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Date;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.sql.DataSource;
+
+/**
+ * Receive data when an investment is sold and updated the investments and participants database.
+ */
+@WebServlet("/sellInvestment")
+public class SellInvestmentServlet extends HttpServlet {
+
+  private static final Logger LOGGER = Logger.getLogger(SellInvestmentServlet.class.getName());
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    long investmentID = -1, priceSold = -1;
```

Comment:
What is price sold? It looks like from the way you are using it that this is the current value of the investment? We have this data in the database, we should be calculating at backend. Perhaps you meant priceSold to represent the amount of the investment they are selling? Are we allowing partial sales?

Reply:
got confused. This data has now been extract from the investments database. (amt_invested)

Reply:
the amount you need to add to amt_available is not amt_invested - it's the amount they cash out. See the investment calculator for functions that work out the current value of an investment. amt_invested is the inital investment, it does not factor in return.

Reply:
I did it from my understanding using the InvestmentCalculator please let me know if it's not right 2aaa2e98d521ea45b07c0423eafd386899d42c80

Reply:
amount available and networth is the same at the moment so I'm definitely doing something wrong.
![image](https://user-images.githubusercontent.com/75593542/107604846-2d8c5d80-6c96-11eb-80d6-45ce5a9eb5b8.png)
But when I used the getInvestmentValue() function i keep getting 0 :(

Reply:
I think this is the main outstanding question of this PR

Reply:
Ok I've done the calculations for the investment before it's sold, add added the price that's sold to netWorth. Want to talk to you before calling the net worth method in the calculator.
23c9fdd1acbdae70761b787d6002e45a1c5615bc

Reply:
Networth doesn't change when you buy or sell, the investment value is already priced in. You just need to update amt_available.

Reply:
Removed ec7ae370c3e1ab3ca4c55d8d08451171ef7788f4